zerver/lib/message.py

```diff
@@ -344,3 +347,103 @@ def render_markdown(message, content, realm=None, realm_alert_words=None, messag
         message.is_me_message = Message.is_status_message(content, rendered_content)
 
     return rendered_content
+
+def add_missing_messages(user_profile):
+    # type: (UserProfile) -> None
+    # This list will store all the messages for which we eventually will create
+    # UserMessage table rows by doing a bulk insert.
+    all_messages_to_insert = []  # type: List[UserMessage]
+
+    def store_user_message_to_insert(message):
+        # type: (Message) -> None
+        message = UserMessage(user_profile=user_profile,
+                              message_id=message['id'], flags=0)
+        all_messages_to_insert.append(message)
+
+    def find_and_store_to_insert_stream_msgs(all_stream_messages,
```

Comment:
If we want to keep this as an inner function, I think we could move it down in the function closer to where it's actually called.  I'd consider just moving it outside the function, so that it doesn't disturb the main flow too much.  You could decouple it from the outer function by just passing in all_messages_to_insert as the list you append to.